orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -394,36 +407,65 @@ def _get_exported_model(self, input_schema, *inputs, **kwargs):
         # correct training flag to reflect the expected behavior.
         # For example, the Dropout node in a model is dropped under eval mode.
         assert self._export_mode is not None, "Please use a concrete instance of ExecutionManager"
-
-        try:
-            with torch.set_grad_enabled(self._enable_custom_autograd_function), \
-                    _logger.suppress_os_stream_output(log_level=self._debug_options.logging.log_level):
-                required_export_kwargs = {'input_names': self._input_info.names,
-                                          'output_names': output_names,
-                                          'opset_version': ortmodule.ONNX_OPSET_VERSION,
-                                          'do_constant_folding': False,
-                                          'training': self._export_mode,
-                                          'dynamic_axes': self._input_info.dynamic_axes,
-                                          'verbose': self._debug_options.logging.log_level < LogLevel.WARNING,
-                                          'export_params': False,
-                                          'keep_initializers_as_inputs': True}
-                invalid_args = self._export_extra_kwargs.keys() & required_export_kwargs.keys()
-                assert len(invalid_args) == 0,\
-                    f"The following PyTorch exporter arguments cannot be specified: '{invalid_args}'."
-                torch.onnx.export(self._flattened_module,
-                                  sample_inputs_as_tuple,
-                                  f,
-                                  **required_export_kwargs,
-                                  **self._export_extra_kwargs)
-        except Exception as e:
-            raise wrap_exception(ORTModuleONNXModelException,
-                                 RuntimeError(f'There was an error while exporting the PyTorch model to ONNX: '
-                                              f'\n\n{_utils.get_exception_as_string(e)}'))
-        exported_model = onnx.load_model_from_string(f.getvalue())
-
-        exported_model = _post_process_after_export(exported_model,
+        provider = self._provider_configs.provider
+        if not provider == "openvino":
+           try:
+               with torch.set_grad_enabled(self._enable_custom_autograd_function), \
```

Comment:
There is so much of reuse between the code if it's openvino or without. I see the main thing that's changing is the export_kwargs, can we create two different maps instead and reuse most of the other code ? It might be cleaner

Reply:
@suryasidd, The export parameters are different for training + evaluation that needs gradient and inference with no gradient logic. It would be easier to make changes going forward with minimal impact to original ORTModule training code having different block for inference.

Reply:
Oh ok. Got it.